### PR TITLE
Migrate builtin types to package types

### PR DESCRIPTION
**Pull request by StachuDotNet**
Besides `HttpClient.Response`, moves the remaining Builtin types to Package space.

|               old builtin name               |                 new package name                 |     notes     |
|----------------------------------------------|--------------------------------------------------|---------------|
|                                              | /packages/process.dark                           |               |
| Builtin.Process.Result                       | @Darklang.Stdlib.Process.Result                  |               |
|                                              |                                                  |               |
|                                              | /packages/darklang/documentation.dark            |               |
| Builtin.DarkInternal.Documentation.Parameter | @Darklang.Documentation.BuiltinFunctionParameter |               |
| Builtin.DarkInternal.Documentation.Function  | @Darklang.Documentation.BuiltinFunction          |               |
|                                              |                                                  |               |
|                                              | /packages/darklang/internal.dark                 |               |
| Builtin.DarkInternal.SchedulingRule.Rule     | @Darklang.Internal.Worker.SchedulingRule         |               |
| Builtin.DarkInternal.Canvas.Program          | @Darklang.Internal.Canvas.Program                |               |
| Builtin.DarkInternal.Canvas.Secret           | @Darklang.Internal.Canvas.Secret                 |               |
| Builtin.DarkInternal.F404.F404               | @Darklang.Internal.Canvas.F404                   | is+was unused |
| Builtin.DarkInternal.Infra.TableSize         | @Darklang.Internal.Infra.TableSize               |               |
|                                              |                                                  |               |
|                                              | /packages/darklang/localexec.dark                |               |
| Builtin.LocalExec.Packages.Function          | @Darklang.LocalExec.Packages.FunctionName        |               |
| Builtin.LocalExec.Packages.Constant          | @Darklang.LocalExec.Packages.ConstantName        |               |
| Builtin.LocalExec.Packages.Type              | @Darklang.LocalExec.Packages.TypeName            |               |
| Builtin.LocalExec.Packages.Package           | @Darklang.LocalExec.Packages.Package             |               |

### CODE REVIEW

File: backend/src/BuiltinDarkInternal/Libs/Documentation.fs [hunk 1]
```diff
@@ -1,4 +1,6 @@
 /// Builtin functions for documentation
+///
+/// TODO: consider moving this from DarkInternal to a publicly-accessible space
```
Comment (StachuDotNet) [inline backend/src/BuiltinDarkInternal/Libs/Documentation.fs:3]: Should users be allowed to call `@Darklang.Documentation.allFunctions`? maybe not immediately useful but seems innocent to allow, to me.

Comment (pbiggar) [reply backend/src/BuiltinDarkInternal/Libs/Documentation.fs:3]: They should not. I would say the opposite: the less surface area to users the better.

**Review by pbiggar — APPROVED** [review]
Looks great!

Should @Darklang.Documentation.BuiltinFunction be internal? Makes more sense to me - I'm not sure what the Documentation namespace is for - from the perspective of a user.

**Comment by StachuDotNet** [comment]
> I'm not sure what the Documentation namespace is for - from the perspective of a user.

I guess this was me prematurely allowing users to create their own devtools.
i.e. "I want to build a little Dark CLI that allows me to quickly check what List functions exist in Darklang"

**Pull request merged**